### PR TITLE
assemble sane no_proxy addresses

### PR DESCRIPTION
**Pull request by areed**
No proxy addresses include localhost, docker0, and master node IP.
Configure NO_PROXY env var for replicated daemon, replicated operator,
and docker on all nodes in a cluster for all schedulers.

### CODE REVIEW

File: install_scripts/templates/replicated-operator.sh [hunk 1]
```diff
@@ -377,6 +380,10 @@ if [ "$NO_PROXY" != "1" ]; then
     if [ -z "$PROXY_ADDRESS" ]; then
         promptForProxy
     fi
+
+    if [ -n "$PROXY_ADDRESS" ]; then
+        getNoProxyAddresses "$DAEMON_ENDPOINT"
```
Comment (emosbaugh) [inline install_scripts/templates/replicated-operator.sh]: should you add private address here as well?

File: install_scripts/templates/replicated-operator.sh [hunk 2]
```diff
@@ -388,15 +395,15 @@ if [ "$SKIP_DOCKER_INSTALL" != "1" ]; then
     checkDockerStorageDriver "$HARD_FAIL_ON_LOOPBACK"
 fi
 
-if [ -n "$PROXY_ADDRESS" ]; then
+if [ "$NO_PROXY" != "1" ] && [ -n "$PROXY_ADDRESS" ]; then
```
Comment (emosbaugh) [inline install_scripts/templates/replicated-operator.sh]: why this change?

File: install_scripts/templates/swarm-worker-join.sh [hunk 3]
```diff
@@ -129,8 +129,7 @@ fi
 promptForSwarmMasterAddress
 
 if [ -n "$PROXY_ADDRESS" ]; then
-    parseIpv4FromAddress "$SWARM_MASTER_ADDRESS"
```
Comment (emosbaugh) [inline install_scripts/templates/swarm-worker-join.sh]: why is this no longer needed? was this a host:port?

File: install_scripts/templates/replicated-operator.sh (hunk 1 above)
Comment (areed) [reply install_scripts/templates/replicated-operator.sh]: On the master the daemon endpoint is the private address and on workers I couldn't think of any reason to add the private address. I guess one possibility would be if someone is running their own private registry in their app, but in that case we'd need to maintain a full list of all IPs in the cluster.
